src/main/java/carpet/script/language/Loops.java

```diff
@@ -140,7 +143,45 @@ public static void apply(Expression expression)
             c.setVariable("_i", _iter);
             return (cc, tt) ->  ret;
         });
+//=========
+        expression.addFunction("zip", (lv) ->
+        {
+            if (lv.stream().allMatch(v->(v instanceof AbstractListValue))){
+                return new LazyListValue(){
+                    List<AbstractListValue> b=lv.stream().map(v->(AbstractListValue)v).collect(Collectors.toList());
+                    List<Iterator<Value>> a=lv.stream().map(v->((AbstractListValue)v).iterator()).collect(Collectors.toList());
+                    
+
+                    @Override
+                    public boolean hasNext() {
+                        return a.stream().allMatch(Iterator::hasNext);
```

Comment:
I would not use streams. zip seems could be used in mission critical situations and could be operating on basic types, like numbers